cairo/tests/utils/args_gen.py

```diff
@@ -1049,6 +1052,12 @@ def _gen_arg(
         if arg_type_origin is tuple and (
             Ellipsis not in get_args(arg_type) or annotations
         ):
+            # Handle conversion from Optimized_Point3D to Optimized_Point2D for BLS12-381
+
+            if arg_type in (Optimized_Point3D[BLSF], Optimized_Point3D[BLSF2]):
+                # arg = normalize(arg)
+                arg = (arg[0], arg[1])
```

Comment:
```suggestion
            if arg_type in (Optimized_Point3D[BLSF], Optimized_Point3D[BLSF2]):
                arg = (arg[0], arg[1])
                assert arg[2] = 1
```
Also assert the point is normalized as a fail safe ?

Reply:
good idea

Reply:
`assert arg[2] = 1` won't work as is, as it must be of type `BLSF` or `BLSF2` but key idea is here